modules/asset-manager-api/src/main/java/org/opencastproject/assetmanager/api/query/RichAResult.java

```diff
@@ -40,36 +42,36 @@ public RichAResult(AResult result) {
     this.result = result;
   }
 
-  public Stream<Property> getProperties() {
-    return result.getRecords().bind(ARecords.getProperties);
+  public List<Property> getProperties() {
+    return ARecords.getProperties(result.getRecords());
   }
 
   /** Count all properties contained in the result. */
   public int countProperties() {
-    return sizeOf(getProperties());
+    return new HashSet(getProperties()).size();
```

Comment:
Since `getProperties` now returns a `List`, can't you just do:
```suggestion
    return getProperties().size();
```